app/assets/javascripts/student_profile/TransitionNotes.js

```diff
@@ -54,52 +54,17 @@ class TransitionNotes extends React.Component {
     this.state = {
       noteText: (regularNote ? regularNote.text : notePrompts),
       restrictedNoteText: (restrictedNote ? restrictedNote.text : restrictedNotePrompts),
-      regularNoteAutosaveStatus: null,
-      restrictedNoteAutosaveStatus: null,
     };
 
-    const throttleOptions = {leading: true, trailing: true};
+    const throttleOptions = {leading: false, trailing: true};
```

Comment:
The default behavior here is `leading: true, trailing: false`, I can't quite follow why we want to change this, can you explain?  Here's what I'm thinking would happen with the default behavior:

```
1. state changes, calls throttled fn
2. throttle calls underlying fn right away
3. other state changes get ignored for 1000ms
4. after 1000ms, then throttle calls underlying fn again
```

Reply:
My thinking was that in step (1), the state has only changed by one character since the user has just started typing.

If we enable `leading`, we're firing off a server call with a single character's worth of data. If we disable `leading`, we let the user type a meaningful word or two and then send that to the server. 

I tried restoring the default `_.throttle` settings locally and (subjectively) found that the overall experience didn't change too much, but did notice an immediate "Saving..." notice as soon as I started typing.

Reply:
Ah gotcha!  Yeah from a "saving all the data" perspective, it does make sense to batch a little bit.  In that case, `debounce` is the function for that I think.  The lodash docs link to a good article with examples about the difference.

Another perspective might be "we want to change everything as soon as possible, but just don't call to the server too much."  That has the nice effect of communicating to the user right away as soon as they change that saving has started.

Thanks for talking this out, either seems 👍 to me!